railties/lib/rails/generators/rails/app/templates/Dockerfile.tt

```diff
@@ -9,15 +9,22 @@ WORKDIR /rails
 
 # Set production environment
 ENV RAILS_ENV="production" \
+    BUNDLE_PATH="vendor/bundle" \
```

Comment:
I'd avoid this, as it tend to mess with naive code that check backtraces paths with `path.start_with?(Rails.root)`.

I mean, not a huge deal, but I don't see the value it brings.

Reply:
Give it a try and check the resulting image sizes.

Reply:
No. Either you make an argument for it, or either I won't bother reviewing your changes.

Reply:
The argument is that I found it to reduce image sizes.  Sorry to be a bother.  Feel free to ignore any or all of these changes.

Reply:
> The argument is that I found it to reduce image sizes.

That's interesting. It's only supposed to install in a different directory, nothing else. I don't see how that would reduce the image size. But ok, you made me curious I'll try it.

Reply:
Ok, so I tested this in a minimal dockefile:

```Dockerfile
FROM ruby:3.2.1
COPY Gemfile Gemfile.lock .
ENV BUNDLE_WITHOUT="development"
RUN bundle install
```

And I get a 1.5MB difference, but in favor of not setting it:

```bash
$ docker history rails-docker-default
IMAGE          CREATED          CREATED BY                                      SIZE      COMMENT
0245ed667651   14 seconds ago   RUN /bin/sh -c bundle install # buildkit        88.6MB    buildkit.dockerfile.v0
<missing>      14 seconds ago   ENV BUNDLE_WITHOUT=development                  0B        buildkit.dockerfile.v0
<missing>      55 seconds ago   COPY Gemfile Gemfile.lock . # buildkit          7.62kB    buildkit.dockerfile.v0
...
```

vs:

```bash
$ docker history rails-docker-vendor-bundle
IMAGE          CREATED              CREATED BY                                      SIZE      COMMENT
43058287772d   51 seconds ago       RUN /bin/sh -c bundle install # buildkit        90.1MB    buildkit.dockerfile.v0
<missing>      51 seconds ago       ENV BUNDLE_PATH=vendor/bundle BUNDLE_WITHOUT…   0B        buildkit.dockerfile.v0
<missing>      About a minute ago   COPY Gemfile Gemfile.lock . # buildkit          7.62kB    buildkit.dockerfile.v0
```

It's a bit unclear why this difference as the trees look very similar.

That being said, digging into the image, there is an easy win that is much more substantial:

<img width="524" alt="Capture d’écran 2023-02-23 à 08 48 59" src="https://user-images.githubusercontent.com/19192189/220849205-660c6320-b3b8-4eda-b6bf-401f01ca39dd.png">

Reply:
Bundler cache also has issues with storing GITHUB_TOKEN secrets when using private repositories. With MRSK, we recommend deleting the cache entirely (https://github.com/mrsked/mrsk#using-build-secrets-for-new-images). We should just do that by default.

Reply:
You link delete another cache that only contains `.gem` files, but yeah we should delete both. I wish bundle had a command or flag to do this. I'll see about making a feature request when I have a bit of time.

Reply:
Ah, the `.bundle/cache` is only in the build image, not in the final, so it's already discarded in a way, we should definitely remove `/usr/local/bundle/cache` though, would save 17MB here.